primitives/staking/src/lib.rs

```diff
@@ -27,3 +28,27 @@ pub type SessionIndex = u32;
 
 /// Counter for the number of eras that have passed.
 pub type EraIndex = u32;
+
+/// Trait describing something that implements a hook for any operations to perform when a staker is
+/// slashed.
+pub trait OnStakerSlash<AccountId, Balance> {
```

Comment:
I wonder if we can solve https://github.com/paritytech/substrate/issues/7681  as well with this new hook.